### ISSUE TITLE
resource/aws_autoscaling_group: Unexpected Removal of Target Group ARNs in version 2.21.0

### ISSUE THREAD

**Issue by JamieKellyBJSS** [terraform-providers/terraform-provider-aws#9513]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
0.11.14

### Affected Resource(s)

* aws_autoscaling_group

### Terraform Configuration Files

3 resources similar to:
```hcl
resource "aws_autoscaling_attachment" "application" {
  autoscaling_group_name = "${module.application.application_asg_name}"
  alb_target_group_arn   = "${aws_alb_target_group.application.arn}"
}
```

### Expected Behavior
When planning using 2.20.0, No infrastructure changes are planned. Expectation is that this remains the case with 2.21.0.

### Actual Behavior
Using 2.21.0 produced the following plan:
```diff
    ~ module.application1.module.application1.aws_autoscaling_group.main
        target_group_arns.#:          "1" => "0"
        target_group_arns.1012598726: "arn:aws:elasticloadbalancing:eu-west-1:xxxxxxxxxxx:targetgroup/application1/3c58ca2c0e49cb1b" => ""
  
    ~ module.application2.module.application2.aws_autoscaling_group.main
        target_group_arns.#:          "2" => "0"
        target_group_arns.3429299186: "arn:aws:elasticloadbalancing:eu-west-1:xxxxxxxxxxx:targetgroup/application2/63ed6bade1545859" => ""
        target_group_arns.703527089:  "arn:aws:elasticloadbalancing:eu-west-1:xxxxxxxxxxx:targetgroup/application2-private/d23ede06f42afcfb" => 
```

### Steps to Reproduce

1. `terraform apply`

**Comment by edmundcraske** [comment]
Not sure if this is caused by #9478 ?

**Comment by pabloajz** [comment]
This is happening also to me since today

**Comment by tcondeixa** [comment]
happening to me too, when you apply multiple time in a row it is alternating between destroying and creating

**Comment by darhon** [comment]
also faced with this problem

**Comment by bflad** [comment]
Hi folks 👋 Sorry for the unexpected new behavior here. 😖

The change within #9478 to start showing the difference of missing configurations of `load_balancers` and `target_group_arns` in the `aws_autoscaling_group` resource was good in isolation, but does not work in combination with the `aws_autoscaling_attachment` resource.

While we sort out the best course of action (potentially reverting the change and cutting a 2.21.1 bug fix release), you have a few options to deal with this situation:

* Update the affected `aws_autoscaling_group` resource configurations with the [`lifecycle` configuration block `ignore_changes` argument](https://www.terraform.io/docs/configuration/resources.html#ignore_changes) available to all resources. This should be used where you are expectedly attaching load balancers outside the `aws_autoscaling_group` resource (e.g. with the `aws_autoscaling_attachment` resource):

```hcl
# Compatible in all versions of Terraform

resource "aws_autoscaling_group" "example" {
  # ... other configuration ...

  lifecycle {
    ignore_changes = ["load_balancers", "target_group_arns"]
  }
}

# Terraform 0.12 and later syntax

resource "aws_autoscaling_group" "example" {
  # ... other configuration ...

  lifecycle {
    ignore_changes = [load_balancers, target_group_arns]
  }
}
```

* Hold back this change in behavior for now without any configuration changes by pinning the Terraform AWS Provider to the previous version, see also the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions), e.g.

```hcl
# Compatible in Terraform 0.10 and later

provider "aws" {
  # ... potentially other configuration ...

  version = "2.20.0"
}

# Terraform 0.12 and later only (applies provider version constraint globally)

terraform {
  # ... potentially other configuration ...

  required_providers {
    # ... potentially other configuration ...

    aws = "2.20.0"
  }
}
```

---

If we revert this change, it is likely to occur again with more proper notice in the next major version release of the Terraform AWS Provider, so if possible, we would recommend the `ignore_changes` configuration change for future proofing.

My apologies again for the unexpected and unannounced change in this behavior. We will update this issue in the next few hours with our plan of action.

**Comment by blalor** [comment]
> If we revert this change, it is likely to occur again with more proper notice in the next major version release of the Terraform AWS Provider, so if possible, we would recommend the `ignore_changes` configuration change for future proofing.

This will be the suggested route forward when using `aws_autoscaling_attachment`? I understand there's no real coordination between resources, but that seems odd to me, lie `aws_autoscaling_attachment` is a 2nd-class resource.

**Comment by while1eq1** [comment]
This also affects terraform version: v0.11.11

**Comment by bflad** [comment]
Please note that we will be cutting a version 2.21.1 of the Terraform AWS Provider shortly with a fix that reverts the change. Associated pull request: https://github.com/terraform-providers/terraform-provider-aws/pull/9518

As mentioned above, this change will likely be redone in a future major version of the Terraform AWS Provider to fix the behavior seen in: https://github.com/terraform-providers/terraform-provider-aws/issues/5197

**Comment by bflad** [comment]
This has been released in [version 2.21.1 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2211-july-26-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests, documentation updates, or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
